packages/model/src/external-author.ts

```diff
@@ -7,3 +7,9 @@ export type ExternalAuthorResponse = Pick<
 >;
 
 export type ListExternalAuthorResponse = ListResponse<ExternalAuthorResponse>;
+
+export type ExternalAuthorPostRequest = {
```

Comment:
I don't think I understand this interface. Like I said earlier it should be either a reference to a user or an external author or a new external author name. In here we seem to always ask for an external author name but also optionally require userId or externalAuthorId?

Reply:
`externalAuthorId` needs to be optional. When selecting an internal user, there's no available an `externalAuthorId`. Below is the interface you suggested. How's this different @peterstarling?

```
authors?: Array<
| { externalAuthorId: string }
| { externalAuthorName: string }
| { userId: string }
```

Reply:
your interface is still invalid. The difference between three optional fields and what I suggested is enormous, it's no surprise that its not visible here though as you don't have tests for the validation either.

You interface cannot allow sending across, for instance, both userId and externalAuthorName in the same object. What would you do in that case? Your code seems to first pick the userId. It should instead throw an error.

So you should:
1. Create a valid interface using unions, as suggested
2. Update your validation to only ever allow ONE property
3. Test the validation